### ISSUE TITLE
训练模型时entity的precision，recall和F1 都是0

### ISSUE THREAD

**Issue by Yuning-J** [ljynlp/W2NER#78]
您好，我使用resume-zh的数据集，然后只改动了.cuda为.cpu，其他按照instruction直接run了main.py。结果所有epoch生成的entity的precision，recall和F1 都是0：
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/transformers/optimization.py:310: FutureWarning: This implementation of AdamW is deprecated and will be removed in a future version. Use the PyTorch implementation torch.optim.AdamW instead, or set `no_deprecation_warning=True` to disable this warning
  FutureWarning,
2022-11-28 22:58:59 - INFO: Epoch: 0
2022-11-29 00:53:08 - INFO: 
+---------+--------+--------+-----------+--------+
| Train 0 |  Loss  |   F1   | Precision | Recall |
+---------+--------+--------+-----------+--------+
|  Label  | 0.3232 | 0.0985 |   0.0999  | 0.0998 |
+---------+--------+--------+-----------+--------+
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
2022-11-29 00:56:13 - INFO: EVAL Label F1 [0.99353707 0.         0.         0.         0.         0.
 0.         0.         0.         0.        ]
2022-11-29 00:56:13 - INFO: 
+--------+--------+-----------+--------+
| EVAL 0 |   F1   | Precision | Recall |
+--------+--------+-----------+--------+
| Label  | 0.0994 |   0.0987  | 0.1000 |
| Entity | 0.0000 |   0.0000  | 0.0000 |
+--------+--------+-----------+--------+
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
2022-11-29 00:59:45 - INFO: TEST Label F1 [0.9937789 0.        0.        0.        0.        0.        0.
 0.        0.        0.       ]
2022-11-29 00:59:45 - INFO: 
+--------+--------+-----------+--------+
| TEST 0 |   F1   | Precision | Recall |
+--------+--------+-----------+--------+
| Label  | 0.0994 |   0.0988  | 0.1000 |
| Entity | 0.0000 |   0.0000  | 0.0000 |
+--------+--------+-----------+--------+

**Comment by ChienWong** [comment]
I found a problem about decode. It will lead to nothing will be decoded.

**Comment by missyoudaisy** [comment]
请问你解决这个问题了吗？

**Comment by songxian1129** [comment]
> 请问你解决这个问题了吗？

请问你解决这个问题了吗

**Comment by songxian1129** [comment]
> 您好，我使用resume-zh的数据集，然后只改动了.cuda为.cpu，其他按照指令直接run了 main.py。结果所有epoch生成的entity的precision，recall和F1 都是0： /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/transformers/optimization.py：310： FutureWarning： 此 AdamW 实现已弃用，并将在将来的版本中删除。请改用 PyTorch 实现 torch.optim.AdamW，或设置为禁用此警告 未来警告， 2022-11-28 22：58：59 - 信息： 纪元： 0 2022-11-29 00：53：08 - 信息： +---------+--------+--------+-----------+--------+ |火车 0 |损失 |F1 |精密 |召回 | +---------+--------+--------+-----------+--------+ |标签 |0.3232 |0.0985 |0.0999 |0.0998 | +---------+--------+--------+-----------+--------+ /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/sklearn/metrics/_classification.py：1318：UndefinedMetricWarning：精度和 F 分数定义不明确，在没有预测样本的标签中设置为 0.0。使用参数来控制此行为。 _warn_prf（平均， 修饰符， msg_start， len（result）） 2022-11-29 00：56：13 - 信息： EVAL Label F1 [0.99353707 0.0. 0.0. 0. 0. 0.0. 0.] 2022-11-29 00：56：13 - 信息：+--------+--------+-----------+--------+ |评估 0 |F1 |精密 |召回 | +--------+--------+-----------+--------+ |标签 |0.0994 |0.0987 |0.1000 | |实体 |0.0000 |0.0000 |0.0000 | +--------+--------+-----------+--------+ /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/sklearn/metrics/_classification.py：1318：UndefinedMetricWarning：精度和 F 分数定义不明确，在没有预测样本的标签中设置为 0.0。使用参数来控制此行为。 _warn_prf（平均， 修饰符， msg_start， len（结果）） 2022-11-29 00：59：45 - 信息：测试标签 F1 [0.9937789 0.0. 0.0. 0.0. 0.0. 0.] 2022-11-29 00：59：45 - 信息：+--------+--------+-----------+--------+ |测试 0 |F1 |精密 |召回 | +--------+--------+-----------+--------+ |标签 |0.0994 |0.0988 |0.1000 | |实体 |0.0000 |0.0000 |0.0000 |`no_deprecation_warning=True``zero_division``zero_division` +--------+--------+-----------+--------+

请问您是否解决了
